File: 737 Gleaming the Cubes/solution.c
#include <stdio.h>

int main() {
	int cubes;
	while(scanf("%d", &cubes) && cubes != 0) {
	    int x, y, z, dist;
	    scanf("%d %d %d %d", &x, &y, &z, &dist);
	    int minX = x, maxX = x + dist, minY = y, maxY = y + dist, minZ = z, maxZ = z + dist;
	    --cubes;
	    while(cubes--) {
	        scanf("%d %d %d %d", &x, &y, &z, &dist);
	        minX = minX > x ? minX : x;
	        maxX = maxX < x + dist ? maxX : x + dist;
	        minY = minY > y ? minY : y;
	        maxY = maxY < y + dist ? maxY : y + dist;
	        minZ = minZ > z ? minZ : z;
	        maxZ = maxZ < z + dist ? maxZ : z + dist;
	    }
	    if(maxX < minX || maxY < minY || maxZ < minZ)
	        puts("0");
	    else
	        printf("%d\n", (maxX - minX) * (maxY - minY) * (maxZ - minZ));
	}
	return 0;
}

File: 737 Gleaming the Cubes/solution.java
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
    	BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StreamTokenizer st = new StreamTokenizer(br);
    	int cubes;
    	StringBuilder output = new StringBuilder();
    	while(st.nextToken() == StreamTokenizer.TT_NUMBER && (cubes = (int)st.nval) != 0) {
    	    st.nextToken();
    	    int x = (int)st.nval;
    	    st.nextToken();
    	    int y = (int)st.nval;
    	    st.nextToken();
    	    int z = (int)st.nval;
    	    st.nextToken();
    	    int dist = (int)st.nval;
    	    int minX = x, maxX = x + dist, minY = y, maxY = y + dist, minZ = z, maxZ = z + dist;
    	    --cubes;
    	    while(cubes-- > 0) {
    	        st.nextToken();
        	    x = (int)st.nval;
        	    st.nextToken();
        	    y = (int)st.nval;
        	    st.nextToken();
        	    z = (int)st.nval;
        	    st.nextToken();
        	    dist = (int)st.nval;
        	    minX = Math.max(minX, x);
        	    maxX = Math.min(maxX, x + dist);
        	    minY = Math.max(minY, y);
        	    maxY = Math.min(maxY, y + dist);
        	    minZ = Math.max(minZ, z);
        	    maxZ = Math.min(maxZ, z + dist);
    	    }
    	    if(maxX < minX || maxY < minY || maxZ < minZ)
	            output.append("0\n");
	        else
	            output.append((maxX - minX) * (maxY - minY) * (maxZ - minZ)).append('\n');
    	}
	    System.out.print(output);
	}
}
